### PR TITLE
Add sorting to grid

### PR DESCRIPTION
**Pull request by cssuh**
Adds sorting ability to the react query results view.

gridSort

### CODE REVIEW

File: src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+export type HeaderFilterCommands = "sort-asc" | "sort-desc";
+
+export interface CommandEventArgs<T extends Slick.SlickData> {
+    grid: Slick.Grid<T>;
+    column: Slick.Column<T>;
+    command: HeaderFilterCommands;
+}
+
+const ShowFilterText = "Show Filter"; //TODO: localize
```
Comment (caohai) [inline src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts:22]: You can do something like this `l10n.t('Show Filter'); `

File: src/reactviews/pages/QueryResult/slickgrid.tsx [hunk 2]
```diff
@@ -180,7 +180,9 @@ const SlickGrid = forwardRef<SlickGridHandle, SlickGridProps>(
 
             let dataProvider = new HybridDataProvider(
                 collection,
-                (_startIndex, _count) => Promise.resolve([]),
+                (_startIndex, _count) => {
+                    return props.loadFunc(_startIndex, _count);
```
Comment (caohai) [inline src/reactviews/pages/QueryResult/slickgrid.tsx:184]: Is this used during sorting?

Comment (caohai) [reply src/reactviews/pages/QueryResult/slickgrid.tsx:184]: I see, no it's fine. Btw you might want to remove the `_` prefix from these parameters as they are being used now.

File: src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        if (offset) {
+            // If there is not enough vertical space under the filter button, we will move up the menu.
+            // const menuTop = offset.top + this.menu.offsetHeight <= window.innerHeight ? offset.top : window.innerHeight - this.menu.offsetHeight;
+            // Make sure the menu is on the screen horizontally.
+            // const menuLeft = offset.left + filterButton.offsetWidth + this.menu.offsetWidth <= window.innerWidth ? offset.left + filterButton.offsetWidth : window.innerWidth - this.menu.offsetWidth;
+        }
+
+        var $popup = jQuery(
+            '<div id="popup-menu">' +
+                '<button id="sort-ascending" type="button" icon="slick-header-menuicon.ascending" class="sort-btn">Sort Ascending</button>' +
```
Comment (caohai) [inline src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts:128]: We need to loc these button text as well

File: src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+            if (
+                !$target.closest("#anchor-btn").length &&
+                !$target.closest("#popup-menu").length
+            ) {
+                $popup.fadeOut();
+            }
+        });
+
+        // Close the pop-up when the close-popup button is clicked
+        jQuery(document).on("click", "#close-popup", function () {
```
Comment (caohai) [inline src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts:160]: Is this `close-popup` button defined/used anywhere?

File: localization/l10n/bundle.l10n.json [hunk 5]
```diff
@@ -99,14 +99,12 @@
   "Move Up": "Move Up",
   "Move Down": "Move Down",
   "Delete": "Delete",
-  "Error loading designer": "Error loading designer",
```
Comment (caohai) [inline localization/l10n/bundle.l10n.json:102]: Do you know why this gets removed?

File: src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts (hunk 4 above)
Comment (cssuh) [reply src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts:161]: Yes, I just added it as an additional option to the menu - good catch.

File: localization/l10n/bundle.l10n.json (hunk 5 above)
Comment (cssuh) [reply localization/l10n/bundle.l10n.json]: Just merged from main, seems like this was removed there.

File: src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+            // const menuTop = offset.top + this.menu.offsetHeight <= window.innerHeight ? offset.top : window.innerHeight - this.menu.offsetHeight;
+            // Make sure the menu is on the screen horizontally.
+            // const menuLeft = offset.left + filterButton.offsetWidth + this.menu.offsetWidth <= window.innerWidth ? offset.left + filterButton.offsetWidth : window.innerWidth - this.menu.offsetWidth;
+        }
+
+        var $popup = jQuery(
+            '<div id="popup-menu">' +
+                `<button id="sort-ascending" type="button" icon="slick-header-menuicon.ascending" class="sort-btn">${l10n.t("Sort Ascending")}</button>` +
+                `<button id="sort-descending" type="button" icon="slick-header-menuicon.descending" class="sort-btn">${l10n.t("Sort Descending")}</button>` +
+                `<button id="close-popup" type="button" class="sort-btn">${l10n.t("Cancel")}</button>` +
```
Comment (caohai) [inline src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts:130]: Ideally the cancel button shouldn't be placed at the same level as the sorting buttons, but I'll let you decide the best way of displaying the button.

Previously we can click other areas to close the pop-up menu, right?

**Review by caohai — APPROVED** [review]

File: src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts (hunk 6 above)
Comment (cssuh) [reply src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts:130]: Yes, we can click other areas to close the pop-up menu here too

Comment (cssuh) [reply src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts:130]: Yeah, we can change the button styling later but I will just merge this in for now

**Pull request merged**